--- SDL-1.2.4/sdl-config.in~	Thu Oct 10 17:46:46 2002
+++ SDL-1.2.4/sdl-config.in	Thu Oct 10 17:46:46 2002
@@ -48,7 +48,7 @@
       echo $includes -I@includedir@/SDL @SDL_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@    --libs|--static-libs)
-@ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      libdirs="-L@libdir@ @SDL_RLD_FLAGS@"
+@ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      libdirs="-L@libdir@
 @ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      echo $libdirs @SDL_LIBS@ @SHARED_SYSTEM_LIBS@
 @ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      ;;
     *)
