--- SDL-1.2.4/sdl-config.in.orig	Wed Mar  6 12:05:47 2002
+++ SDL-1.2.4/sdl-config.in	Tue Apr 23 20:42:12 2002
@@ -47,15 +47,10 @@
       fi
       echo $includes -I@includedir@/SDL @SDL_CFLAGS@
       ;;
-@ENABLE_SHARED_TRUE@    --libs)
-@ENABLE_SHARED_TRUE@      libdirs="-L@libdir@ @SDL_RLD_FLAGS@"
-@ENABLE_SHARED_TRUE@      echo $libdirs @SDL_LIBS@ @SHARED_SYSTEM_LIBS@
-@ENABLE_SHARED_TRUE@      ;;
-@ENABLE_STATIC_TRUE@@ENABLE_SHARED_TRUE@    --static-libs)
-@ENABLE_STATIC_TRUE@@ENABLE_SHARED_FALSE@    --libs|--static-libs)
-@ENABLE_STATIC_TRUE@      libdirs="-L@libdir@ @SDL_RLD_FLAGS@"
-@ENABLE_STATIC_TRUE@      echo $libdirs @SDL_LIBS@ @STATIC_SYSTEM_LIBS@
-@ENABLE_STATIC_TRUE@      ;;
+@ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@    --libs|--static-libs)
+@ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      libdirs="-L@libdir@ @SDL_RLD_FLAGS@"
+@ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      echo $libdirs @SDL_LIBS@ @SHARED_SYSTEM_LIBS@
+@ENABLE_SHARED_TRUE@@ENABLE_STATIC_TRUE@      ;;
     *)
       echo "${usage}" 1>&2
       exit 1
