--- SDL-1.2.5/src/main/Makefile.am.orig	Sun May 19 22:06:01 2002
+++ SDL-1.2.5/src/main/Makefile.am	Sun Oct 13 21:04:54 2002
@@ -14,15 +14,11 @@
 # Build a separate library containing the main() entry point.
 lib_LIBRARIES = libSDLmain.a
 
-if TARGET_MACOSX
-MAINLIB_ARCH_SRCS = SDLMain.m SDLMain.h
-else
 if TARGET_QTOPIA
 MAINLIB_ARCH_SRCS = SDL_Qtopia_main.cc
 else
 MAINLIB_ARCH_SRCS = SDL_main.c
 endif
-endif
 libSDLmain_a_SOURCES = $(MAINLIB_ARCH_SRCS)
 
 # Build an internal library of any special app setup functions
--- SDL-1.2.5/configure.in.orig	Sun Oct 13 20:59:37 2002
+++ SDL-1.2.5/configure.in	Sun Oct 13 21:06:35 2002
@@ -2548,18 +2549,6 @@
 CXXFLAGS="$CFLAGS"
 
 
-# Check for darwin at the very end and set up the Objective C compiler
-# We do this here so that we get the full CFLAGS into OBJCFLAGS
-case "$target" in
-    *-*-darwin*)
-        OBJC="cc"
-        OBJCFLAGS="$CFLAGS"
-        OBJCDEPMODE="$CCDEPMODE"
-        AC_SUBST(OBJC)
-        AC_SUBST(OBJCFLAGS)
-        AC_SUBST(OBJCDEPMODE)
-        ;;
-esac
 
 # Finally create all the generated files
 dnl Important: Any directory that you want to be in the distcheck should
@@ -2613,7 +2602,6 @@
 src/video/maccommon/Makefile
 src/video/macdsp/Makefile
 src/video/macrom/Makefile
-src/video/quartz/Makefile
 src/video/svga/Makefile
 src/video/vgl/Makefile
 src/video/aalib/Makefile
