diff -Nru SDL-1.2.4.orig/configure.in SDL-1.2.4/configure.in
--- SDL-1.2.4.orig/configure.in	Sun Apr 14 01:57:24 2002
+++ SDL-1.2.4/configure.in	Wed Jul 31 10:04:28 2002
@@ -1414,6 +1414,7 @@
     fi
 }
 
+AM_CONDITIONAL(USE_DIRECTX,false)
 case "$target" in
     *-*-linux*)
         ARCH=linux
@@ -2416,7 +2417,6 @@
         OBJCDEPMODE="$CCDEPMODE"
         AC_SUBST(OBJC)
         AC_SUBST(OBJCFLAGS)
-        AC_SUBST(OBJCDEPMODE)
         ;;
 esac
 
