--- SDL-1.2.7/sdl-config.in.orig	2004-02-18 18:21:55.000000000 +0100
+++ SDL-1.2.7/sdl-config.in	2004-02-27 23:00:09.602808160 +0100
@@ -51,12 +51,12 @@
       #echo $includes -I@includedir@/SDL @SDL_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@    --libs)
-@ENABLE_SHARED_TRUE@      libdirs="-L@libdir@ @SDL_RLD_FLAGS@"
+@ENABLE_SHARED_TRUE@      libdirs="-L@libdir@"
 @ENABLE_SHARED_TRUE@      echo $libdirs @SDL_LIBS@ @SHARED_SYSTEM_LIBS@
 @ENABLE_SHARED_TRUE@      ;;
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_TRUE@    --static-libs)
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_FALSE@    --libs|--static-libs)
-@ENABLE_STATIC_TRUE@      libdirs="-L@libdir@ @SDL_RLD_FLAGS@"
+@ENABLE_STATIC_TRUE@      libdirs="-L@libdir@"
 @ENABLE_STATIC_TRUE@      echo $libdirs @SDL_STATIC_LIBS@ @STATIC_SYSTEM_LIBS@
 @ENABLE_STATIC_TRUE@      ;;
     *)
