--- SDL-1.2.6/configure.in.orig	2003-09-05 20:48:37.000000000 +0200
+++ SDL-1.2.6/configure.in	2003-09-05 21:10:36.881227832 +0200
@@ -336,11 +336,11 @@
             esd_lib_spec=`echo $ESD_LIBS | sed 's/.*-L\([[^ ]]*\).*/\1\/libesd.so.*/'`
             esd_lib=`ls $esd_lib_spec | head -1 | sed 's/.*\/\(.*\)/\1/'`
             echo "-- $esd_lib_spec -> $esd_lib"
-            if test x$enable_dlopen != xyes && \
+            if test x$enable_sdlopen != xyes && \
                test x$enable_esd_shared = xyes; then
-                AC_MSG_ERROR([You must have dlopen() support and use the --enable-dlopen option])
+                AC_MSG_ERROR([You must have dlopen() support and use the --enable-sdlopen option])
             fi
-            if test x$enable_dlopen = xyes && \
+            if test x$enable_sdlopen = xyes && \
                test x$enable_esd_shared = xyes && test x$esd_lib != x; then
                 CFLAGS="$CFLAGS -DESD_SUPPORT -DESD_DYNAMIC=\$(esd_lib) $ESD_CFLAGS"
                 AC_SUBST(esd_lib)
@@ -386,11 +386,11 @@
                 arts_lib_spec=`echo $ARTSC_LIBS | sed 's/.*-L\([[^ ]]*\).*/\1\/libartsc.so.*/'`
                 arts_lib=`ls $arts_lib_spec | head -1 | sed 's/.*\/\(.*\)/\1/'`
                 echo "-- $arts_lib_spec -> $arts_lib"
-                if test x$enable_dlopen != xyes && \
+                if test x$enable_sdlopen != xyes && \
                    test x$enable_arts_shared = xyes; then
-                    AC_MSG_ERROR([You must have dlopen() support and use the --enable-dlopen option])
+                    AC_MSG_ERROR([You must have dlopen() support and use the --enable-sdlopen option])
                 fi
-                if test x$enable_dlopen = xyes && \
+                if test x$enable_sdlopen = xyes && \
                    test x$enable_arts_shared = xyes && test x$arts_lib != x; then
                     CFLAGS="$CFLAGS -DARTSC_SUPPORT -DARTSC_DYNAMIC=\$(arts_lib) $ARTSC_CFLAGS"
                     AC_SUBST(arts_lib)
@@ -1557,10 +1557,10 @@
 dnl Check for the dlfcn.h interface for dynamically loading objects
 CheckDLOPEN()
 {
-    AC_ARG_ENABLE(dlopen,
-[  --enable-dlopen         use dlopen for shared object loading [default=no]],
-                  , enable_dlopen=yes)
-    if test x$enable_dlopen = xyes; then
+    AC_ARG_ENABLE(sdlopen,
+[  --enable-sdlopen         use dlopen for shared object loading [default=no]],
+                  , enable_sdlopen=yes)
+    if test x$enable_sdlopen = xyes; then
         AC_MSG_CHECKING(for dlopen)
         use_dlopen=no
         AC_TRY_COMPILE([
